codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -65,14 +65,26 @@ public DefaultHttp2LocalFlowController(Http2Connection connection,
         connection.addListener(new Http2ConnectionAdapter() {
             @Override
             public void onStreamAdded(Http2Stream stream) {
-                stream.localFlowState(new DefaultFlowState(stream, 0));
+                // Unconditionally used the reduced flow control state because it requires no object allocation
+                // and the DefaultFlowState will be allocated in onStreamActive.
+                stream.localFlowState(REDUCED_FLOW_STATE);
```

Comment:
oh my mistake, it won't since it's a static variable. In the remote flow controller case, it will.